lib/npm-check-updates.js

```diff
@@ -188,6 +188,11 @@ function analyzeProjectDependencies(options, pkgData, pkgFile) {
 
     print(options, `Fetching ${vm.getVersionTarget(options)} versions...`, 'verbose');
 
+    if (options.enginesNode) {
+        options.enginesNode = _.get(pkg, 'engines.node');
+        options.versionTarget = 'latest';
```

Comment:
Can you explain forcing versionTarget to `latest`? Do other version targets not work with `--engines-node`?

Reply:
Because my idea was: 
if I enable engines matching, I want to get exactly latest version.

I can try to implement engines matching to all targets, but I think this will be crazy changes...

Reply:
I can't justifiably add this feature only for `latest` when it should work for all version targets. This will require some extra code, but I trust you will be able to abstract out redundant code appropriately. If you are unsure the best approach architecturally, feel free to ask questions or request feedback on an initial approach. Thank you!

Reply:
@raineorshine How about this?
- minimum extra code
- works with all targets